codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -63,9 +63,11 @@
     private final IntObjectMap<Http2Stream> streamMap = new IntObjectHashMap<Http2Stream>();
     private final ConnectionStream connectionStream = new ConnectionStream();
     private final Set<Http2Stream> activeStreams = new LinkedHashSet<Http2Stream>();
+    private final Queue<ActiveStreamsEvent> activeStreamsPendingEvents = new ArrayDeque<ActiveStreamsEvent>(4);
     private final DefaultEndpoint<Http2LocalFlowController> localEndpoint;
     private final DefaultEndpoint<Http2RemoteFlowController> remoteEndpoint;
     private final Http2StreamRemovalPolicy removalPolicy;
+    private int activeStreamsIterationCount;
```

Comment:
@Scottmitch 

I wonder if it makes sense to move all this active streams related stuff into its own class? Something like

```java
    private static class ActiveStreams {
        LinkedHashSet<Http2Stream> streams;
        Queue<ActiveStreamsEvent> pendingEvents = new ArrayDeque<ActiveStreamsEvent>();
        int pendingIterations;
        
        Http2Stream forEachActiveStream(ActiveStreamVisitor visitor) {
        ...
        }

        interface Event {
            void process();
        }
    }
```